azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java

```diff
@@ -41,11 +41,11 @@ public void addKeyVaultPropertySource() {
                 this.environment.getProperty(Constants.AZURE_KEYVAULT_REFRESH_INTERVAL))
                 .map(Long::valueOf).orElse(Constants.DEFAULT_REFRESH_INTERVAL_MS);
         final long timeAcquiringTimeoutInSeconds = this.environment.getProperty(
-            Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, 60L);
+            Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, Constants.TOKEN_ACQUIRE_TIMEOUT_SECS);
         final ServiceClientCredentials credentials;
 
-        if (this.environment.containsProperty(Constants.AZURE_MSI_ENDPOINT)
-            && this.environment.containsProperty(Constants.AZURE_MSI_SECRET)) {
+        if (this.environment.containsProperty("MSI_ENDPOINT")
+            && this.environment.containsProperty("MSI_SECRET")) {
```

Comment:
Shouldn't these be properties that are passed in by the client?  The way it's written right now I could configure it to use a client-id and client-secret and if the environment variable MSI_ENDPOINT and MSI_SECRET exist it would use MSI instead.  Seems misleading to me

Reply:
No, these properties aren't passed by the client. When you host the app on Azure App service environment, it will be picked up from there if MSI is enabled in the App service.

Yes, that is right, the way it is written is if you have MSI enabled, always use that else fallback to the usual client secret authentication. That is the expected behavior as per my understanding.